Assets/RedBlueGames/BulkRename/Editor/BulkRenamerWindow.cs

```diff
@@ -211,64 +210,33 @@ private void OnGUI()
             EditorGUILayout.EndScrollView();
         }
 
-        private string[] GetNamesFromSelections()
+        private void SortObjectsToRename()
         {
-
-            List<string> allAssetNames = new List<string>();
-            List<Object> allSortedGameObectsNames = new List<Object>();
-
-            int namesCount = this.objectsToRename.Count;
-            var names = new string[namesCount];
-            bool isItAsset = AssetDatabase.GetAssetPath(objectsToRename[0]) != "";
-
-
-            #region Sorting GameObjects in Hierarchy Window
-            if (!isItAsset)
+            // Nothing to sort if objects to rename haven't been initialized
+            if (this.objectsToRename == null)
             {
-                allSortedGameObectsNames = objectsToRename.OrderBy(x => x.name.Length).ThenBy(x => x.name).ToList();
+                return;
             }
-            #endregion
-            #region Sorting Assets in Project Window
-            if (isItAsset)
-            {
-                string path = "";
-                int startIndex = 0;
-                int endIndex = 0;
-                for (int i = 0; i < namesCount; ++i)
-                {
-                    path = AssetDatabase.GetAssetPath(objectsToRename[i]);
-                    startIndex = path.LastIndexOf('/') + 1;
-                    endIndex = path.LastIndexOf('.');
 
-                    string extractedNameOfAsset = path.Substring(startIndex, endIndex - startIndex);
```

Comment:
Quick note - you can use `System.IO.Path.GetFilename` to achieve this same thing without the string parsing.